projects/ngx-sub-form/src/lib/ngx-sub-form.component.ts

```diff
@@ -188,24 +200,36 @@ export abstract class NgxSubFormComponent<ControlInterface, FormInterface = Cont
 
     const transformedValue: FormInterface = this.transformToFormGroup(obj);
 
-    // for now we throw an error if the transformed value isn't an object with all the expect values
-    // there's an issue to track support for an array https://github.com/cloudnc/ngx-sub-form/issues/9
-    if (Array.isArray(transformedValue)) {
-      throw new ArrayNotTransformedBeforeWriteValueError();
-    }
-
     const missingKeys: (keyof FormInterface)[] = this.getMissingKeys(transformedValue);
     if (missingKeys.length > 0) {
       throw new MissingFormControlsError(missingKeys as string[]);
     }
 
+    this.handleFormArrayControls(transformedValue);
+
     this.formGroup.setValue(transformedValue, {
       emitEvent: false,
     });
     this.formGroup.markAsPristine();
     this.formGroup.markAsUntouched();
   }
 
+  private handleFormArrayControls(obj: any) {
+    Object.entries(obj).forEach(([key, value]) => {
+      if (this.formGroup.get(key) instanceof FormArray && Array.isArray(value)) {
+        const formArray: FormArray = this.formGroup.get(key) as FormArray;
+
+        while (formArray.length > value.length) {
+          formArray.removeAt(formArray.length - 1);
+        }
+
+        while (formArray.length < value.length) {
+          formArray.push(new FormControl());
```

Comment:
We should have an overridable hook that allows developers to override this constructor, in case they want to have a custom validator on the control. Something like

```ts
type KeysWithType<T, V> = {
  [K in keyof T]: T[K] extends V ? K : never
}[keyof T];

type ArrayPropertyOf<T> = KeysWithType<T, Array<any>>;



  protected createFormArrayControl(key: ArrayPropertyOf<FormInterface>): FormControl {

    switch (key) {
      case 'foo':
        return new FormControl(null, Validators.required);
      default:
        return; //maybe `undefined` can be a default behavior we decide like `new FormControl(null)` 
    }

  }


```

Reply:
Nice one! I've used it :+1: